h2o-py/tests/testdir_apis/Data_Manipulation/pyunit_h2oH2OFrame_match.py

```diff
@@ -4,28 +4,72 @@
 from tests import pyunit_utils
 from h2o.utils.typechecks import assert_is_type
 from h2o.frame import H2OFrame
-from random import randrange
-import numpy as np
+import math
 
 
 def h2o_H2OFrame_match():
     """
     Python API test: h2o.frame.H2OFrame.match(table, nomatch=0)
 
-    Copied from runit_lstrip.R
     """
     iris = h2o.import_file(path=pyunit_utils.locate("smalldata/iris/iris.csv"))
-    matchFrame = iris["C5"].match(['Iris-setosa'])
-    assert_is_type(matchFrame, H2OFrame)    # check return type
-    assert matchFrame.sum()[0,0]==50.0, "h2o.H2OFrame.match() command is not working."  # check return result
 
-    matchFrame = iris["C5"].match(['Iris-setosa', 'Iris-versicolor'])
-    assert_is_type(matchFrame, H2OFrame)    # check return type
-    assert matchFrame.sum()[0,0]==100.0, "h2o.H2OFrame.match() command is not working."  # check return result
+    nomatch = 0
+    match_frame = iris["C5"].match(['Iris-setosa'], nomatch=nomatch)
+    assert_is_type(match_frame, H2OFrame)    # check return type
+    assert match_frame.sum()[0, 0] == 50.0, "h2o.H2OFrame.match() command is not working."  # check return result
+    assert match_frame[0, 0] == 1, "match value should be 1"
+    assert match_frame[50, 0] == nomatch, "match value should be 0"
+    assert match_frame[100, 0] == nomatch, "match value should be 0"
 
-    matchFrame = iris["C5"].match(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'])
-    assert_is_type(matchFrame, H2OFrame)    # check return type
-    assert matchFrame.sum()[0,0]==150.0, "h2o.H2OFrame.match() command is not working."  # check return result
+    match_frame = iris["C5"].match(['Iris-setosa', 'Iris-versicolor'], nomatch=nomatch)
+    assert_is_type(match_frame, H2OFrame, nomatch=0)    # check return type
+    assert round(match_frame.sum()[0, 0]) == 150, "h2o.H2OFrame.match() command is not working."  # check return result
+    assert match_frame[0, 0] == 1, "match value should be 1"
+    assert match_frame[50, 0] == 2, "match value should be 2"
+    assert match_frame[100, 0] == nomatch, "match value should be 0"
 
+    match_values = ['Iris-setosa', 'Iris-versicolor', 'Iris-virginica']
```

Comment:
can you please add a test with duplicate values to be sure?
```match_values = match_values + match_values``` ?
result should be the same if I'm correct